Migrate to newer googletest verion

We have to update the googletest framework since GCC > 7.1 throws a warning with the current tests:
1.8.1 + 1.10.0 do not compile for C++:

Compilation failed (exit status 1):
/home/travis/build/Yakindu/statecharts/googletest-release-1.8.1/googlemock/build/libgmock_main.a(gmock_main.cc.o): In function `main':
gmock_main.cc:(.text+0x31): undefined reference to `testing::InitGoogleMock(int*, char**)'
collect2: error: ld returned 1 exit status
	at org.yakindu.sct.generator.cpp.test.OperationsTest.setUp(OperationsTest.java:33)

--- a/test-plugins/org.yakindu.sct.generator.c.test/src/org/yakindu/sct/generator/c/gtest/GTestHelper.java
+++ b/test-plugins/org.yakindu.sct.generator.c.test/src/org/yakindu/sct/generator/c/gtest/GTestHelper.java
@@ -216,6 +216,7 @@ public class GTestHelper {
 				//.wPedantic()
 			 	.wAll()				
 			 	.wExtra()
+			 	.wConversion()
 			 	.wError();
 		 if(compiler == Compiler.GCC) {
 			 gTestCommand.wnoUnusedParameter(); // ignore unused parameter flag

--- a/test-plugins/org.yakindu.sct.generator.c.test/src/org/yakindu/sct/generator/c/gtest/GTestHelper.java
+++ b/test-plugins/org.yakindu.sct.generator.c.test/src/org/yakindu/sct/generator/c/gtest/GTestHelper.java
@@ -220,8 +220,6 @@ public class GTestHelper {
 		 if(compiler == Compiler.GCC) {
 			 gTestCommand.wnoUnusedParameter(); // ignore unused parameter flag
 			 gTestCommand.wnoUnusedFunction(); // ignore unused functions flag
-			 
-			 gTestCommand.wConversion(); // TODO: Only activated for C; does currently not work with GMock and Cpp (see #2903)
 		 }
 		return gTestCommand.build();
 	}

--- a/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ChildFirstOrthogonalReactions/ChildFirstOrthogonalReactionsTest.cc
+++ b/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ChildFirstOrthogonalReactions/ChildFirstOrthogonalReactionsTest.cc
@@ -4,11 +4,6 @@
 #include "ChildFirstOrthogonalReactions.h"
 #include "sc_runner.h"
 #include "sc_types.h"
-#include "gmock/gmock.h"
-
-using ::testing::AtLeast;
-using ::testing::Return;
-using ::testing::_;
 
 namespace  {
 

--- a/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ChildFirstTransitionsinSubchart/ChildFirstTransitionsinSubchartTest.cc
+++ b/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ChildFirstTransitionsinSubchart/ChildFirstTransitionsinSubchartTest.cc
@@ -4,11 +4,6 @@
 #include "ChildFirstTransitionsInSubchart.h"
 #include "sc_runner.h"
 #include "sc_types.h"
-#include "gmock/gmock.h"
-
-using ::testing::AtLeast;
-using ::testing::Return;
-using ::testing::_;
 
 namespace  {
 

--- a/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/OperationsTest/OperationsTest.cc
+++ b/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/OperationsTest/OperationsTest.cc
@@ -4,11 +4,6 @@
 #include "Operations.h"
 #include "sc_runner.h"
 #include "sc_types.h"
-#include "gmock/gmock.h"
-
-using ::testing::AtLeast;
-using ::testing::Return;
-using ::testing::_;
 
 namespace  {
 

--- a/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ParentFirstOrthogonalReactions/ParentFirstOrthogonalReactionsTest.cc
+++ b/test-plugins/org.yakindu.sct.generator.cpp.test/gtests/ParentFirstOrthogonalReactions/ParentFirstOrthogonalReactionsTest.cc
@@ -4,11 +4,6 @@
 #include "ParentFirstOrthogonalReactions.h"
 #include "sc_runner.h"
 #include "sc_types.h"
-#include "gmock/gmock.h"
-
-using ::testing::AtLeast;
-using ::testing::Return;
-using ::testing::_;
 
 namespace  {
 

